src/executionFiles/bridges/nxtp.execute.ts

```diff
@@ -219,15 +220,35 @@ export class NXTPExecutionManager {
         (transfer) =>
           transfer.crosschainTx.invariant.transactionId === transactionId
       )
+
       if (foundTransaction) {
-        claimProcess.txHash = foundTransaction.fulfilledTxHash
-        claimProcess.txLink =
-          toChain.metamask.blockExplorerUrls[0] + 'tx/' + claimProcess.txHash
-        claimProcess.message = 'Swapped:'
-        status.fromAmount = estimate.fromAmount
-        status.toAmount = estimate.toAmount
-        status.status = 'DONE'
-        setStatusDone(update, status, claimProcess)
+        switch (foundTransaction.status) {
+          case 'CANCELLED':
+            claimProcess.message =
+              'CANCELLED - Funds have been refunded on source chain.'
```

Comment:
Do we know they have already been refunded or may this take a while?

Reply:
If it is in this state the refund has already been happen, sadly `foundTransaction` does not contain a hash of the cancel/refund transaction. Would be good to show it.